src/data/resolvers/coinMachine.ts

```diff
@@ -99,41 +99,47 @@ export const coinMachineResolvers = ({
         return null;
       }
     },
-    async coinMachineBoughtTokens(_, { colonyAddress }) {
+    async coinMachineBoughtTokens(_, { colonyAddress, transactionHash }) {
+      const { provider } = colonyManager.networkClient;
       try {
         const coinMachineClient = await colonyManager.getClient(
           ClientType.CoinMachineClient,
           colonyAddress,
         );
 
-        const subgraphData = await apolloClient.query<
-          SubgraphCoinMachinePeriodsQuery,
-          SubgraphCoinMachinePeriodsQueryVariables
-        >({
-          query: SubgraphCoinMachinePeriodsDocument,
-          variables: {
-            colonyAddress: colonyAddress.toLowerCase(),
-            extensionAddress: coinMachineClient.address.toLowerCase(),
-            /*
-             * The limit doesn't matter in this case as we're not using that data,
-             * we're just interested in the TokensBought events
-             */
-            limit: 1,
-          },
-          fetchPolicy: 'network-only',
-        });
+        const boughtTokensFilter = coinMachineClient.filters.TokensBought(
+          null,
```

Comment:
The `buyer` argument is sadly not `indexed`, otherwise we could have filtered by that too _(which would have been amazing)_